### ISSUE TITLE
Tests not respecting maxParallelForks?

### ISSUE THREAD

**Issue by yissachar** [scoverage/gradle-scoverage#60]
I'm using Spark and running `reportScoverage` results in all sorts of concurrency errors. 

I have https://github.com/maiflai/gradle-scalatest setup and initially running `gradle test` gave me similar issues with Spark until I set:

```
test {
    maxParallelForks = 1
}
```
Now I can run `gradle test` and it completes fine. However, it seems that this plugin does not respect maxParallelForks, since I am still getting the thread errors even with it set.

Is there any way to tell it not to run tests in parallel? Alternatively can I tell it to skip running tests since I already have those from `gradle test`?

**Comment by maiflai** [comment]
Hi,

I think the snippet posted only configures the specific task called **test** - this plugin defines an additional `Test` task called **testScoverage**.

If you want to configure all `Test` tasks then you might try the following?

```groovy
tasks.withType(Test) {
    maxParallelForks = 1
}
```

The reason for defining a new task is that it runs against a separate, instrumented, set of **main** classes, and it's easier to manage the up-to-date state of a `Task` that way.

If you always want to run tests only against the instrumented classes, then you might consider modifying the **test** task using `upToDateWhen`, and then configure the build such that **reportScoverage** is run (either by explicitly naming the task or by configuring `dependsOn` appropriately).

Hope this helps?

Stu

**Comment by yissachar** [comment]
Brilliant! That snippet worked perfectly for me. Thanks for the help and for both plugins!

**Comment by atronchi** [comment]
Should we not update the parallel testing section of the README with the above snippet?
https://github.com/maiflai/gradle-scalatest#parallel-testing
Otherwise it seems to no longer work since 0.13...

**Comment by maiflai** [comment]
The two are separate I think - the README you're quoting is the gradle-scalatest README, and it doesn't create new tasks.

It doesn't always follow that you want to run every test task in full parallel, and in my private experience I've found that it helps to balance this based on the parallelisation of the compile tasks.

How do you mean "no longer work" please?

**Comment by atronchi** [comment]
Sorry about the missing context: I had to upgrade several projects from 0.11 and 0.13 to 0.14 this morning to fix https://github.com/maiflai/gradle-scalatest/issues/37
All these projects were configured with ```
test {
    maxParallelForks = 1
}
```
After the upgrade, this setting seems to be no longer respected and I get errors during testing: ```17/08/29 12:51:51 ERROR ScalaTest-1 SparkContext: Error initializing SparkContext.
java.net.BindException: Can't assign requested address: Service 'sparkDriver' failed after 16 retries! Consider explicitly setting the appropriate port for the service 'sparkDriver' (for example spark.ui.
port for SparkUI) to an available port or increasing spark.port.maxRetries.```
These typically occur only when multiple sparkContexts have been initialized, i.e. when multiple tests run in parallel.

**Comment by maiflai** [comment]
Ah, ok. 

Please could you try upgrading to the current version (0.16)?

If that doesn't work can we move this to the gradle-scalatest repository and open a new issue there?

Thanks,
Stu

**Comment by atronchi** [comment]
Will do. Thanks!

**Comment by atronchi** [comment]
OK, still failing with 0.16. :\

**Comment by maiflai** [comment]
No worries. Ideally we could work over the issue with access to your project.

If not, please could you submit a build log with --debug enabled?

**Comment by atronchi** [comment]
Will do, and re-open in an issue for gradle-scalatest. Thanks!
